app/models/user.rb

```diff
@@ -7,6 +7,7 @@ class User < ActiveRecord::Base
   validates :name, presence: true
   validates :email, presence: true, uniqueness: true
   validates :password, presence: true, on: :create, length: { minimum: 7 }
+  before_create :generate_token
```

Comment:
Your app uses it at that point, however your database can be stolen, and whoever has that token can just enter it to password reset url and edit password for the user with that token. If you create a user and automatically generate that token it sits there and waits to be used. Making it very vulnerable part of your app. Expiration would have helped it as well.